File: component-acgnews/src/main/java/com/rabtman/acgnews/api/AcgNewsService.java
package com.rabtman.acgnews.api;

import com.rabtman.acgnews.base.constant.HtmlConstant;
import com.rabtman.acgnews.mvp.model.entity.SHPage;
import com.rabtman.acgnews.mvp.model.entity.SHPostDetail;
import com.rabtman.acgnews.mvp.model.entity.SHResponse;
import io.reactivex.Flowable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author Rabtman
 */
public interface AcgNewsService {

  @GET(HtmlConstant.ISH_URL + "ver/1f590097/article/list")
  Flowable<SHResponse<SHPage>> getISHNews(@Query("page") int pageIndex,
      @Query("pageSize") int pageSize, @Query("categoryID") int categoryID);

  @GET(HtmlConstant.ISH_URL + "ver/3c590236/article/detail")
  Flowable<SHResponse<SHPostDetail>> getISHNewsDetail(@Query("id") int postId);
}

File: component-acgnews/src/main/java/com/rabtman/acgnews/api/AcgNewsService.kt
package com.rabtman.acgnews.api

import com.rabtman.acgnews.base.constant.HtmlConstant
import com.rabtman.acgnews.mvp.model.entity.SHPage
import com.rabtman.acgnews.mvp.model.entity.SHPostDetail
import com.rabtman.acgnews.mvp.model.entity.SHResponse
import io.reactivex.Flowable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Rabtman
 */
interface AcgNewsService {
    @GET(HtmlConstant.ISH_URL + "ver/1f590097/article/list")
    fun getISHNews(@Query("page") pageIndex: Int,
                   @Query("pageSize") pageSize: Int, @Query("categoryID") categoryID: Int): Flowable<SHResponse<SHPage>>

    @GET(HtmlConstant.ISH_URL + "ver/3c590236/article/detail")
    fun getISHNewsDetail(@Query("id") postId: Int): Flowable<SHResponse<SHPostDetail>>
}